libs/cassandra-util/src/Cassandra/Schema.hs

```diff
@@ -182,6 +203,49 @@ migrateSchema l o ms = do
     metaInsert :: QueryString W (Int32, Text, UTCTime) ()
     metaInsert = "insert into meta (id, version, descr, date) values (1,?,?,?)"
 
+-- | Retrieve and compare local and peer system schema versions.
+-- if they don't match, retry once per second for 30 seconds
+waitForSchemaConsistency :: Client ()
+waitForSchemaConsistency = do
+    void $ retryWhileN 30 inDisagreement getSystemVersions
+  where
+    getSystemVersions :: Client (UUID, [UUID])
+    getSystemVersions = do
+        -- These two sub-queries must be made to the same node.
+        -- (comparing local from node A and peers from node B wouldn't be correct)
+        -- using the custom 'migrationPolicy' when connecting to cassandra ensures this.
+        local <- systemLocalVersion
+        peers <- systemPeerVersions
+        case local of
+            Just localVersion -> return $ (localVersion, peers)
+            Nothing           -> liftIO $ throwIO $ ErrorCall
+                "No system_version in system.local (should not happen)"
+
+    inDisagreement :: (UUID, [UUID]) -> Bool
+    inDisagreement (localVersion, peers) = not $ all (== localVersion) peers
+
+    systemPeerVersions :: Client [UUID]
+    systemPeerVersions = fmap runIdentity <$> qry
+      where
+        qry = retry x1 (query cql (params One ()))
+
+        cql :: PrepQuery R () (Identity UUID)
+        cql = "select schema_version from system.peers"
+
+    systemLocalVersion :: Client (Maybe UUID)
```

Comment:
Can you switch the order of these definitions to be the same as the one that's used when calling them? (local, peer)